src/coreclr/gc/gc.cpp

```diff
@@ -23683,6 +23683,7 @@ void gc_heap::mark_phase (int condemned_gen_number, BOOL mark_only_p)
         grow_mark_list_piece();
 #endif //USE_REGIONS
 
+        GCToEEInterface::BeforeGcScanRoots(max_generation, /* is_bgc */ false, /* is_concurrent */ false);
```

Comment:
this should be `condemned_gen_number` instead of max_generation as we are not necessarily collecting max_generation

Reply:
Boo. Thanks. I completely missed this.